src/main/java/org/jabref/logic/citationstyle/CitationStyle.java

```diff
@@ -115,36 +120,51 @@ public static CitationStyle createCitationStyleFromFile(final String styleFile)
         return null;
     }
 
+    /**
+     * Provides the default citation style which is currently IEEE
+     * @return default citation style
+     */
     public static CitationStyle getDefault() {
         return createCitationStyleFromFile(DEFAULT);
     }
 
     /**
-     * THIS ONLY WORKS WHEN JabRef IS STARTED AS AN APPLICATION (JAR)
-     *
-     * Reads all available CitationStyle in the Jar
+     * Provides the citation styles that come with JabRef.
+     * @return list of available citation styles
      */
     public static List<CitationStyle> discoverCitationStyles() {
+        if (!STYLES.isEmpty()) {
+            return STYLES;
+        }
         try {
-            final List<CitationStyle> citationStyles = new ArrayList<>();
-            String path = CitationStyle.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath();
-
-            try (JarFile file = new JarFile(path)) {
-                Enumeration<JarEntry> entries = file.entries();
-                while (entries.hasMoreElements()) {
-                    String filename = entries.nextElement().getName();
-                    if (!filename.startsWith("dependent") && filename.endsWith("csl")) {
-                        CitationStyle citationStyle = CitationStyle.createCitationStyleFromFile(filename);
-                        if (citationStyle != null) {
-                            citationStyles.add(citationStyle);
-                        }
+
+            Path filePath = Paths.get(CitationStyle.class.getProtectionDomain().getCodeSource().getLocation().toURI());
```

Comment:
@halirutan The Paths.get method has an overload for accepting an URI. 
The misleading method of the URI class (getPath) does not work  as itdoesn't inlcude the "scheme" part of the URI object which is used to identify the underlying FileSytem implementation.
This tutorial has a nice overview of "what is a path" https://docs.oracle.com/javase/tutorial/essential/io/path.html

If you happen to use a Classpath url or sth else, the recommended way is the same  Paths.get(Class.toUrl.toUri)

Reply:
That wasn't my code :) When I see this right, it was coming from Christian Bartsch but it doesn't matter. Good that you spotted it.